### PR TITLE
feat: add type declaration

### PR DESCRIPTION
**Pull request by LoneRifle**
- port type declaration from @DefinitelyTyped, and reference it from package.json
- add definitions for KeyInfo, StringKeyInfo, tweak SignedXml to take a KeyInfo for `keyInfoProvider` instead of FileKeyInfo

### CODE REVIEW

**Comment by cjbarth** [comment]
I am on a special project for about 4 more weeks, so I'll be delayed in anything GitHub-related until after then.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    ): void;
+    getOriginalXmlWithIds(): string;
+    getSignatureXml(): string;
+    getSignedXml(): string;
+    loadSignature(signatureNode: string | Node): void;
+}
+
+export interface KeyInfo {
+    getKey(keyInfo?: Node): Buffer;
```
Comment (djaqua) [inline index.d.ts:80]: Ha! I brought this issue up to the DT maintainer: getKey should be accepting an array of Nodes, not a single Node object

I will link the issue and explanation here in a moment

**Review by djaqua — CHANGES_REQUESTED** [review]

**Comment by djaqua** [comment]
Save for the one comment I left, this looks great!

File: index.d.ts (hunk 1 above)
Comment (LoneRifle) [reply index.d.ts:80]: So something like...
```suggestion
    getKey(keyInfo?: null | Node[]): Buffer;
```
Per that discussion?

**Review by djaqua — APPROVED** [review]
Awesome 🙌

**Review by cjbarth — APPROVED** [review]

**Pull request merged**